### PR TITLE
REACT-MP-2020 task#5 form

### PR DESCRIPTION
**Pull request by garnich**
This PR include:

Task 7: (Form)

### CODE REVIEW

File: src/components/modalWindows/addMovie/addMovie.js [hunk 1]
```diff
@@ -1,151 +1,193 @@
...
-        // --------------------------
-      }
 
-      handleClose();
-    }, [])
+    const formik = useFormik({
+        initialValues: {
+          ...movie
+        },
+        validationSchema: Yup.object({
```
Comment (DenisKarlyuk) [inline src/components/modalWindows/addMovie/addMovie.js]: Move the validation schema to separate file `move.schema.js`

File: src/components/modalWindows/addMovie/addMovie.js [hunk 2]
```diff
@@ -1,151 +1,193 @@
...
+          poster_path: Yup.string().url().required('Wrong URL'),
+          overview: Yup.string()
+            .max(500, 'Must be 15 characters or less')
+            .required('Required'),
+          runtime: Yup.number()
+            .positive()
+            .integer()
+            .required('Required'),
+        }),
+        onSubmit: values => {
```
Comment (DenisKarlyuk) [inline src/components/modalWindows/addMovie/addMovie.js]: `onSubmit` should be a `redux` action or , you can add `editMode` to `initialValues` and get it in the action.
```javascript
const handleSubmit = (values) =>
  handleChangeMovie(values) // <-- redux action
    .then(handleClose)

File: src/components/modalWindows/addMovie/addMovie.js [hunk 3]
```diff
@@ -1,151 +1,193 @@
...
               <p 
                 name="id"
                 id="id">
-                {id}
+                {activeId}
               </p>
             </Fragment>
           )}
           <label htmlFor="title">Title</label>
           <input
```
Comment (DenisKarlyuk) [inline src/components/modalWindows/addMovie/addMovie.js:51]: https://formik.org/docs/api/field
The best way is to use a wrapper, it isn't necessary. This example from my project:
```javascript
import React, { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import { useField, useFormikContext } from 'formik';
import { Box, FormHelperText } from '@material-ui/core';
import FormLabel from '../FormLabel';

import useStyles from './formikField.styles';

const isEvent = (event) => event && (event instanceof Event || event.nativeEvent instanceof Event);

const FormikField = ({
  title,
  direction,
  alignItems,
  component: Component,
  name,
  helperText,
  my,
  mb,
  mt,
  width,
  ...props
}) => {
  const { submitCount } = useFormikContext();
  const [field, meta, helper] = useField(name);
  const { helperText: helperTextStyles, secondLine } = useStyles();
  const { setValue, setTouched } = helper;
  const { onChange, onBlur } = field;
  const { error, touched } = meta;

  const handleChange = useCallback(async (eventOrValue) => {
    if (isEvent(eventOrValue)) {
      onChange(eventOrValue);
    } else {
      await setValue(eventOrValue, false);
      setTouched(true, true);
    }
  }, [onChange, setTouched, setValue]);

  const handleBlur = useCallback((eventOrValue) => {
    if (isEvent(eventOrValue)) {
      onBlur(eventOrValue);
    } else {
      setTouched(true);
    }
  }, [onBlur, setTouched]);

  const isError = !!error && (touched || !!submitCount);

  return (
    <Box
      display="flex"
      flexDirection={direction}
      alignItems={alignItems}
      mb={mb || my}
      mt={mt || my}
    >
      { title && (
        <Box mb={direction === 'row' ? 0 : 1} mr={1}>
          <FormLabel title={title} />
        </Box>
      )}
      <Box flexGrow={1} position="relative">
        <Box width={width}>
          <Component
            {...props}
            {...field}
            onChange={handleChange}
            onBlur={handleBlur}
            error={isError || null}
            name={name}
          />
        </Box>
        <FormHelperText
          className={helperTextStyles}
          error={isError && !helperText}
        >
          {!helperText && isError ? error : helperText}
        </FormHelperText>
        {helperText && isError && (
          <FormHelperText
            className={`${helperTextStyles} ${secondLine}`}
            error={isError}
          >
            {error}
          </FormHelperText>
        )}
      </Box>
    </Box>
  );
};

FormikField.defaultProps = {
  direction: 'column',
  alignItems: 'stretch',
  title: '',
  helperText: '',
  my: 3,
  mb: 0,
  mt: 0,
  width: 'auto',
};

FormikField.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.shape({}),
  ]).isRequired,
  name: PropTypes.string.isRequired,
  direction: PropTypes.string,
  title: PropTypes.string,
  alignItems: PropTypes.string,
  helperText: PropTypes.string,
  my: PropTypes.number,
  mb: PropTypes.number,
  mt: PropTypes.number,
  width: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
};

export default memo(FormikField);

File: src/components/modalWindows/deleteMovie/deleteMovie.js [hunk 4]
```diff
@@ -38,6 +40,12 @@ const DeleteMovie = (props) => {
   )
 }
 
+const mapDispatchToProps = (dispatch) => {
+  return {
+    deleteMovie: (id) => {dispatch(removeMovie(id))},
```
Comment (DenisKarlyuk) [inline src/components/modalWindows/deleteMovie/deleteMovie.js]: You can use short form 
```javascript
const mapDispatchToProps = {
  deleteMovie: removeMovie,
};

File: src/components/movieCardSorter/movieCardSorter.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { sortMovies } from './../../actions/actions';
 
 const MovieCardSorter = (props) => {
   const options = [
```
Comment (DenisKarlyuk) [inline src/components/movieCardSorter/movieCardSorter.js]: Move it out of function scope

File: src/reducer/reducer.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+} from './../constants/constants';
+
+import { 
+    updateObjectInArray,
+    removeObjectFromArray,
+    movieSoreterAndFilter
+} from './../utils/helper';
+
+const reducer = (state, action) => {
+    if(state === undefined) {
```
Comment (DenisKarlyuk) [inline src/reducer/reducer.js]: You can use something like that
```javascript
const initialState = { ... };
const reducer = (state = initialState, action) => ...

Comment (garnich) [reply src/reducer/reducer.js]: done

File: src/components/movieCardSorter/movieCardSorter.js (hunk 5 above)
Comment (garnich) [reply src/components/movieCardSorter/movieCardSorter.js]: done

File: src/components/modalWindows/deleteMovie/deleteMovie.js (hunk 4 above)
Comment (garnich) [reply src/components/modalWindows/deleteMovie/deleteMovie.js]: done

File: src/components/modalWindows/addMovie/addMovie.js (hunk 3 above)
Comment (garnich) [reply src/components/modalWindows/addMovie/addMovie.js:51]: OK, thanks.

File: src/components/modalWindows/addMovie/addMovie.js (hunk 1 above)
Comment (garnich) [reply src/components/modalWindows/addMovie/addMovie.js]: done

File: src/components/modalWindows/addMovie/addMovie.js (hunk 2 above)
Comment (garnich) [reply src/components/modalWindows/addMovie/addMovie.js]: done

Comment (garnich) [reply src/components/modalWindows/addMovie/addMovie.js]: done

**Pull request merged**